guides/process/mentoring/supporting_and_mentoring_other_engineers.md

```diff
@@ -0,0 +1,63 @@
+# Supporting and Mentoring other Engineers
```

Comment:
I think there should be a section on the patience of mentors. 

I think it's highly important for mentors to create a sense of comfort for their mentee, this means removing pressure.